2024-07-11 12:33:33.024341: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2024-07-11 12:33:33.024519: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2024-07-11 12:33:41,417 | app.py:165 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2024-07-11 12:33:41,520 | app.py:179 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2024-07-11 12:33:41,520 | server.py:89 | Initializing global parameters
INFO flwr 2024-07-11 12:33:41,520 | server.py:276 | Requesting initial parameters from one random client
INFO flwr 2024-07-11 12:33:42,831 | server.py:280 | Received initial parameters from one random client
INFO flwr 2024-07-11 12:33:42,831 | server.py:91 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:33:42,847 | server.py:98 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.45}
INFO flwr 2024-07-11 12:33:42,848 | server.py:104 | FL starting
DEBUG flwr 2024-07-11 12:33:42,999 | server.py:227 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2024-07-11 12:33:43,028 | server.py:241 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2024-07-11 12:33:43,030 | fedavg.py:242 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:33:43,039 | server.py:131 | fit progress: (1, 0.42163859064167286, {'accuracy': 0.79}, 0.19124810000000103)
DEBUG flwr 2024-07-11 12:33:43,039 | server.py:178 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:33:43,061 | server.py:192 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2024-07-11 12:33:43,061 | fedavg.py:273 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2024-07-11 12:33:43,062 | server.py:227 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:33:43,081 | server.py:241 | fit_round 2 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:33:43,092 | server.py:131 | fit progress: (2, 0.4245968896201889, {'accuracy': 0.79}, 0.24387360000000058)
DEBUG flwr 2024-07-11 12:33:43,092 | server.py:178 | evaluate_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:33:43,109 | server.py:192 | evaluate_round 2 received 3 results and 0 failures
DEBUG flwr 2024-07-11 12:33:43,109 | server.py:227 | fit_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:33:43,128 | server.py:241 | fit_round 3 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:33:43,138 | server.py:131 | fit progress: (3, 0.4245960756695948, {'accuracy': 0.79}, 0.2903435999999999)
DEBUG flwr 2024-07-11 12:33:43,139 | server.py:178 | evaluate_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:33:43,156 | server.py:192 | evaluate_round 3 received 3 results and 0 failures
DEBUG flwr 2024-07-11 12:33:43,156 | server.py:227 | fit_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:33:43,186 | server.py:241 | fit_round 4 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:33:43,197 | server.py:131 | fit progress: (4, 0.4246902881089279, {'accuracy': 0.7866666666666666}, 0.34887830000000086)
DEBUG flwr 2024-07-11 12:33:43,197 | server.py:178 | evaluate_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:33:43,216 | server.py:192 | evaluate_round 4 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:33:43,217 | server.py:227 | fit_round 5: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:33:43,243 | server.py:241 | fit_round 5 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:33:43,255 | server.py:131 | fit progress: (5, 0.42470755293004087, {'accuracy': 0.7866666666666666}, 0.4069769999999995)
DEBUG flwr 2024-07-11 12:33:43,256 | server.py:178 | evaluate_round 5: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:33:43,277 | server.py:192 | evaluate_round 5 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:33:43,278 | server.py:227 | fit_round 6: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:33:43,305 | server.py:241 | fit_round 6 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:33:43,316 | server.py:131 | fit progress: (6, 0.42472542909880884, {'accuracy': 0.7866666666666666}, 0.4686111999999998)
DEBUG flwr 2024-07-11 12:33:43,317 | server.py:178 | evaluate_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:33:43,339 | server.py:192 | evaluate_round 6 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:33:43,339 | server.py:227 | fit_round 7: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:33:43,364 | server.py:241 | fit_round 7 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:33:43,376 | server.py:131 | fit progress: (7, 0.4260928960029489, {'accuracy': 0.7933333333333333}, 0.5276087999999994)
DEBUG flwr 2024-07-11 12:33:43,376 | server.py:178 | evaluate_round 7: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:33:43,399 | server.py:192 | evaluate_round 7 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:33:43,400 | server.py:227 | fit_round 8: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:33:43,432 | server.py:241 | fit_round 8 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:33:43,443 | server.py:131 | fit progress: (8, 0.42611316519012976, {'accuracy': 0.7933333333333333}, 0.594834500000001)
DEBUG flwr 2024-07-11 12:33:43,443 | server.py:178 | evaluate_round 8: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:33:43,468 | server.py:192 | evaluate_round 8 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:33:43,468 | server.py:227 | fit_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:33:43,500 | server.py:241 | fit_round 9 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:33:43,512 | server.py:131 | fit progress: (9, 0.426976118207546, {'accuracy': 0.79}, 0.6643029000000009)
DEBUG flwr 2024-07-11 12:33:43,512 | server.py:178 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:33:43,538 | server.py:192 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:33:43,539 | server.py:227 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:33:43,578 | server.py:241 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:33:43,590 | server.py:131 | fit progress: (10, 0.42698726235682366, {'accuracy': 0.79}, 0.7426636999999996)
DEBUG flwr 2024-07-11 12:33:43,591 | server.py:178 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:33:43,617 | server.py:192 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2024-07-11 12:33:43,617 | server.py:153 | FL finished in 0.769463
INFO flwr 2024-07-11 12:33:43,618 | app.py:225 | app_fit: losses_distributed [(1, 0.41422308422625065), (2, 0.4128111768513918), (3, 0.41281016170978546), (4, 0.412906954685847), (5, 0.4129109670718511), (6, 0.41515402779692695), (7, 0.41485083387011573), (8, 0.4148517939306441), (9, 0.41868689571108136), (10, 0.4186854839324951)]
INFO flwr 2024-07-11 12:33:43,618 | app.py:226 | app_fit: metrics_distributed_fit {}
INFO flwr 2024-07-11 12:33:43,618 | app.py:227 | app_fit: metrics_distributed {}
INFO flwr 2024-07-11 12:33:43,618 | app.py:228 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.42163859064167286), (2, 0.4245968896201889), (3, 0.4245960756695948), (4, 0.4246902881089279), (5, 0.42470755293004087), (6, 0.42472542909880884), (7, 0.4260928960029489), (8, 0.42611316519012976), (9, 0.426976118207546), (10, 0.42698726235682366)]
INFO flwr 2024-07-11 12:33:43,618 | app.py:229 | app_fit: metrics_centralized {'accuracy': [(0, 0.45), (1, 0.79), (2, 0.79), (3, 0.79), (4, 0.7866666666666666), (5, 0.7866666666666666), (6, 0.7866666666666666), (7, 0.7933333333333333), (8, 0.7933333333333333), (9, 0.79), (10, 0.79)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
>>>>>>>>>>>>> evaluating on server-side
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.88364211,  0.85056232,  0.75597826,  0.52929068,  0.29257815,
         0.1900984 ,  0.0140426 ,  0.04694142]]), array([-0.19342501])]
[array([[-1.88364211,  0.85056232,  0.75597826,  0.52929068,  0.29257815,
         0.1900984 ,  0.0140426 ,  0.04694142]]), array([-0.19342501])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.91636893,  0.91271327,  0.82968066,  0.52015154,  0.18970263,
         0.16447588,  0.02268   , -0.01376677]]), array([-0.14164035])]
[array([[-1.91636893,  0.91271327,  0.82968066,  0.52015154,  0.18970263,
         0.16447588,  0.02268   , -0.01376677]]), array([-0.14164035])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.91652165,  0.91295371,  0.82992371,  0.51997825,  0.18972533,
         0.16455425,  0.0225795 , -0.01367789]]), array([-0.14196836])]
[array([[-1.91652165,  0.91295371,  0.82992371,  0.51997825,  0.18972533,
         0.16455425,  0.0225795 , -0.01367789]]), array([-0.14196836])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.94721187,  0.90223215,  0.87627772,  0.43583764,  0.18604935,
         0.09582967,  0.02961148,  0.02265951]]), array([0.00522776])]
[array([[-1.94721187,  0.90223215,  0.87627772,  0.43583764,  0.18604935,
         0.09582967,  0.02961148,  0.02265951]]), array([0.00522776])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.94612453,  0.90149872,  0.87545568,  0.43566636,  0.18575474,
         0.09584041,  0.02947939,  0.02265477]]), array([0.00644318])]
[array([[-1.94612453,  0.90149872,  0.87545568,  0.43566636,  0.18575474,
         0.09584041,  0.02947939,  0.02265477]]), array([0.00644318])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.94480725,  0.90074716,  0.87462731,  0.43546933,  0.18539624,
         0.09577655,  0.02934775,  0.02273232]]), array([0.00737167])]
[array([[-1.94480725,  0.90074716,  0.87462731,  0.43546933,  0.18539624,
         0.09577655,  0.02934775,  0.02273232]]), array([0.00737167])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.89875565,  0.92862206,  0.83238373,  0.42375225,  0.19892125,
         0.0730592 , -0.00313588,  0.03888715]]), array([0.01715832])]
[array([[-1.89875565,  0.92862206,  0.83238373,  0.42375225,  0.19892125,
         0.0730592 , -0.00313588,  0.03888715]]), array([0.01715832])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.89883224,  0.9286397 ,  0.83204406,  0.42370829,  0.19869763,
         0.072813  , -0.00349951,  0.03871196]]), array([0.01865428])]
[array([[-1.89883224,  0.9286397 ,  0.83204406,  0.42370829,  0.19869763,
         0.072813  , -0.00349951,  0.03871196]]), array([0.01865428])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.89450354,  0.9257455 ,  0.80104564,  0.41330075,  0.19353843,
         0.055227  , -0.01766423,  0.01573037]]), array([0.07187374])]
[array([[-1.89450354,  0.9257455 ,  0.80104564,  0.41330075,  0.19353843,
         0.055227  , -0.01766423,  0.01573037]]), array([0.07187374])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.89635399,  0.92643676,  0.80074254,  0.41355718,  0.19341477,
         0.05508445, -0.01816489,  0.01538184]]), array([0.07485598])]
[array([[-1.89635399,  0.92643676,  0.80074254,  0.41355718,  0.19341477,
         0.05508445, -0.01816489,  0.01538184]]), array([0.07485598])]
>>>>>>>>>>>>>run time:
0.515625s
